libclc/ptx-nvidiacl/libspirv/group/collectives_helpers.ll

```diff
@@ -0,0 +1,61 @@
+; 64 storage locations is sufficient for all current-generation NVIDIA GPUs
+; 64 bits per warp is sufficient for all fundamental data types
+; Reducing storage for small data types or increasing it for user-defined types
+; will likely require an additional pass to track group algorithm usage
+@__clc__group_scratch = internal addrspace(3) global [64 x i64] undef, align 1
+
+define i8 addrspace(3)* @__clc__get_group_scratch_bool() nounwind alwaysinline {
+entry:
+  %ptr = getelementptr inbounds [64 x i64], [64 x i64] addrspace(3)* @__clc__group_scratch, i64 0, i64 0
+  %cast = bitcast i64 addrspace(3)* %ptr to i8 addrspace(3)*
+  ret i8 addrspace(3)* %cast
+}
+
+define i8 addrspace(3)* @__clc__get_group_scratch_char() nounwind alwaysinline {
+entry:
+  %ptr = getelementptr inbounds [64 x i64], [64 x i64] addrspace(3)* @__clc__group_scratch, i64 0, i64 0
+  %cast = bitcast i64 addrspace(3)* %ptr to i8 addrspace(3)*
+  ret i8 addrspace(3)* %cast
+}
+
+define i8 addrspace(3)* @__clc__get_group_scratch_short() nounwind alwaysinline {
```

Comment:
i8 -> i16?

Reply:
Fixed in 3375433c974f4446409e4585754f97abb8e6ab59.